### ISSUE TITLE
Reproducibility  of interpretation.ipynb

### ISSUE THREAD

**Issue by raytrun** [peterant330/CLIP_AFT#1]
I'm running the interpretation.ipynb, and the model parameters were loaded from [vit_l_14. pt.](https://mycuhk-my.sharepoint.com/:u:/g/personal/1155187960_link_cuhk_edu_hk/EfUVcY67_xdIi-Q9sfBNWhEBiPAeY3Zu-uAo4XyUdqk2wQ?e=IlEVfy) 
However, I've observed significant discrepancies in both the Gradients Map and Grad-CAM compared to the original results. 
This line was the only addition. Am I reproducing it incorrectly?

```python
model.visual.load_state_dict(torch.load("vit_l_14.pt", map_location=torch.device('cpu')))
``` 

The original notebook output is on the left, and my results are on the right.

<table>
  <tr>
    <td><img src="https://github.com/user-attachments/assets/78f2b4de-02af-4cfd-9280-d2c46ac41c50" width="400"></td>
    <td><img src="https://github.com/user-attachments/assets/c0f6dddb-4201-4384-bcdc-b707ebc90e9a" width="400"></td>
  </tr>
  <tr>
    <td><img src="https://github.com/user-attachments/assets/4521df28-f0e8-48c1-9657-6665aed7d961" width="400"></td>
    <td><img src="https://github.com/user-attachments/assets/240d9b02-9100-4079-a2a3-5ff6396a1eca" width="400"></td>
  </tr>
</table>

**Comment by peterant330** [comment]
Hi, you may try ViT-B-16 instead.

**Comment by raytrun** [comment]
Thank you for your response. 
Comparing original CLIP (left) and your finetuned CLIP (right) on ViT-B-16 using Grad-CAM 1.5.5.
The Grad-CAM visualizations I reproduced didn’t look very good—could this be due to the version of Grad-CAM I used? May I ask which version you are using?

<table>
  <tr>
    <td><img src="https://github.com/user-attachments/assets/95499122-23f7-4bc4-bd12-1694cf765579" width="400"></td>
    <td><img src="https://github.com/user-attachments/assets/14e98e8b-c782-4f69-9026-cb96a8b4ed94" width="400"></td>
  </tr>
  <tr>
    <td><img src="https://github.com/user-attachments/assets/61cdd51f-31ae-4d99-b0f3-4996fb959397" width="400"></td>
    <td><img src="https://github.com/user-attachments/assets/c0e51806-fda6-4717-b45e-10cd0d97173e" width="400"></td>
  </tr>

**Comment by peterant330** [comment]
This is a known phenomenon that the Grad-CAM of CLIP sometimes flip the foreground and background. For such cases we would take the reverse of the Grad-CAM for better visualization.
